clients/src/main/java/org/apache/kafka/common/requests/ProduceResponse.java

```diff
@@ -266,15 +266,14 @@ protected Struct toStruct(short version) {
                         .set(PARTITION_ID, partitionEntry.getKey())
                         .set(ERROR_CODE, errorCode)
                         .set(BASE_OFFSET_KEY_NAME, part.baseOffset);
-                if (partStruct.hasField(LOG_APPEND_TIME_KEY_NAME))
-                    partStruct.set(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
+                partStruct.setIfExists(LOG_APPEND_TIME_KEY_NAME, part.logAppendTime);
                 partStruct.setIfExists(LOG_START_OFFSET_FIELD, part.logStartOffset);
 
                 List<Struct> errorRecords = new ArrayList<>();
```

Comment:
Similarly, we can initialize this lazily. This potentially has a substantial benefit on the broker since produce requests are on the hot path.